### PR TITLE
vulture: init at 1.6

### PR DESCRIPTION
**Pull request by mcwitt**
###### Motivation for this change

Adds the [vulture](https://github.com/jendrikseipp/vulture) tool for detecting dead code in Python.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by davidak — CHANGES_REQUESTED** [review]
Thanks for your contribution! The code looks good.

It seems dependencies are missing:

```
ERROR: could not install deps [coverage, pytest, pytest-cov]; v = InvocationError(u'/build/vulture-1.6/.tox/py/bin/python -m pip install coverage pytest pytest-cov', 1)

3 packages failed to build:
python27Packages.vulture python37Packages.vulture python38Packages.vulture
```

https://gist.github.com/davidak/71903b64737e4d30a6e287c754c1e3ac

**Comment by mcwitt** [comment]
@davidak thanks for catching that. I hadn't yet tried to build with sandboxing (but now realize that is super important!). I think the issue was trying to run the tests with `tox` (which does some impure stuff trying to create virtualenvs and pip-install test dependencies).

I found a [couple](https://github.com/mcwitt/nixpkgs/blob/63cacba96ef68921ee7a2a18ed2af5b7f0937b3c/pkgs/development/python-modules/pytest-factoryboy/default.nix#L27) of [examples](https://github.com/mcwitt/nixpkgs/blob/63cacba96ef68921ee7a2a18ed2af5b7f0937b3c/pkgs/development/python-modules/west/default.nix#L25) of packages using `tox` where we either fall back `pytest` or disable checks altogether. For now I opted to use `pytest` directly.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -7697,6 +7697,7 @@ in {
 
   rxv     = callPackage ../development/python-modules/rxv     { };
 
+  vulture = callPackage ../development/python-modules/vulture { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:7700]: please try to add this in a semi-sorted fashion. Appending to the end will cause merge conflicts to others who do the same.

File: pkgs/development/python-modules/vulture/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+buildPythonPackage rec {
+  pname = "vulture";
+  version = "1.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1sbwbwkpk3s7iwnwsdrvj1ydw9lgbn3xqhji7f8y5y6vvr77i53v";
+  };
+
+  buildInputs = [ setuptools tox ];
```
[the remark this answers is not in the document]
Comment (mcwitt) [reply pkgs/development/python-modules/vulture/default.nix]: Fixed in 63cacba. I ended up removing `tox` altogether and using `pytest` (https://github.com/NixOS/nixpkgs/pull/94396#issuecomment-667591019).

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (mcwitt) [reply pkgs/top-level/python-packages.nix]: Fixed in 0674589.

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -5181,6 +5181,12 @@
     githubId = 13689192;
     name = "Nguyễn Gia Phong";
   };
+  mcwitt = {
+    email = "mcwitt@gmail.com";
+    github = "mcwitt";
+    githubId = 319411;
+    name = "Matt Wittmann";
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix:5189]: Please make this into a separate commit:

```
maintainers: add mcwitt
```

**Review by jonringer — COMMENTED** [review]
should be two commits:

```
maintainers: add mcwitt
pythonPackages.vulture: init at 1.6
```

File: maintainers/maintainer-list.nix (hunk 3 above)
Comment (mcwitt) [reply maintainers/maintainer-list.nix:5189]: Done in d988033.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 94396 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>3 packages built:</summary>  - python27Packages.vulture  - python37Packages.vulture  - python38Packages.vulture</details>
